subprojects/base-services/src/main/java/org/gradle/util/TextUtil.java

```diff
@@ -61,8 +62,16 @@ public static String getPlatformLineSeparator() {
     /**
      * Converts all line separators in the specified string to the specified line separator.
      */
-    public static String convertLineSeparators(String str, String sep) {
-        return str == null ? null : str.replaceAll("\r\n|\r|\n", sep);
+    @Nullable
+    public static String convertLineSeparators(@Nullable String str, String sep) {
+        return str == null ? null : replaceLineSeparatorsOf(str, sep);
+    }
+
+    /**
+     * Converts all line separators in the specified non-null string to the specified line separator.
+     */
+    public static String replaceLineSeparatorsOf(CharSequence string, String bySeparator) {
+        return Pattern.compile("\r\n|\r|\n").matcher(string).replaceAll(bySeparator);
     }
```

Comment:
Doing this ends up replacing correct separators, too (with the same separator), resulting in a lot of unnecessary string concatenation. In most cases what we want is `convertLineSeparatorsToUnix()` where we could be more opinionated and not match `\n` at all. WDYT?

Reply:
Thanks, @lptr. That's a good idea, will do 👍

Reply:
@lptr, I've gone a little further and made it avoid all the allocations when the `\r` character doesn't appear in the input string.

Reply:
I don't think you need the explicit `indexOf()` call—if the `replaceAll()` regex doesn't match, it will return the original string.

What you could do to speed things up a bit and avoid unnecessary work is to pre-compile the `Pattern`.

Reply:
Right, it would return the original string after compiling the pattern and doing at least one unnecessary allocation.

I'm ambivalent on this one so I'll go with your suggestion and just precompiled the pattern. The code is simpler that  way. Thanks.